node/cn/gasprice/gasprice_test.go

```diff
@@ -17,14 +17,92 @@
 package gasprice
 
 import (
+	"context"
 	"math/big"
 	"testing"
 
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/blockchain/vm"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/math"
+	"github.com/klaytn/klaytn/consensus/gxhash"
+	"github.com/klaytn/klaytn/crypto"
+	"github.com/klaytn/klaytn/networks/rpc"
+	"github.com/klaytn/klaytn/params"
+	"github.com/klaytn/klaytn/storage/database"
+
 	"github.com/golang/mock/gomock"
 	mock_api "github.com/klaytn/klaytn/api/mocks"
 	"github.com/stretchr/testify/assert"
 )
 
+const testHead = 32
+
+type testBackend struct {
+	chain *blockchain.BlockChain
+}
+
+func (b *testBackend) HeaderByNumber(ctx context.Context, number rpc.BlockNumber) (*types.Header, error) {
+	if number > testHead {
+		return nil, nil
+	}
+	if number == rpc.LatestBlockNumber {
+		number = testHead
+	}
+	return b.chain.GetHeaderByNumber(uint64(number)), nil
+}
+
+func (b *testBackend) BlockByNumber(ctx context.Context, number rpc.BlockNumber) (*types.Block, error) {
+	if number > testHead {
+		return nil, nil
+	}
+	if number == rpc.LatestBlockNumber {
+		number = testHead
+	}
+	return b.chain.GetBlockByNumber(uint64(number)), nil
+}
+
+func (b *testBackend) GetBlockReceipts(ctx context.Context, hash common.Hash) types.Receipts {
+	return b.chain.GetReceiptsByBlockHash(hash)
+}
+
+func (b *testBackend) ChainConfig() *params.ChainConfig {
+	return b.chain.Config()
+}
+
+func newTestBackend(t *testing.T) *testBackend {
```

Comment:
Just curious. Why did you implement this here instead of `node/cn/gasprice/feehistory_test.go`?

Reply:
This is based on Etheruem's PR.
That testBackend is used by not only fee history tests but also other tests like (SuggestTipCap) in Ethereum.

Currently, the only place where the test backend is used is the fee history test, 
but in case test cases requiring testBackend will be added later, 
the location is separated like Ethereum.